tools/ovc/openvino/tools/ovc/moc_frontend/pipeline.py

```diff
@@ -90,156 +92,164 @@ def check_places_are_same(places_original: List[Place], places_new: List[Place])
-                except NotImplementedFailure:
-                    raise Error("Please specify type for value freezing {} node explicitly "
-                                "because the frontend does not support automatic type detection.".format(name))
-                # in case of cutting graph (or using custom inputs) and unspecified or dynamic type,
-                # the default type is fp32
-                if ov_type == Type.undefined or ov_type == Type.dynamic:
-                    ov_type = Type.f32
-                dtype = get_numpy_ctype(ov_type)
-
-            input_model.set_element_type(place, ov_type)
-            # prepare and cast value to dtype
-            if isinstance(value, list):
-                casted_list = list()
-                for v in mo_array(value):
-                    casted_list.append(np_map_cast[dtype](v))
-                value = mo_array(casted_list, dtype=dtype)
+                    model.add_name_for_tensor(
+                        new_input['node'], new_input['input_name'])
+                except NotImplementedFailure as e:
+                    # some frontends might not implement this method
+                    log.warning('Could not add an additional name to a tensor pointed to by \'{}\'. Details: {}'.format(
+                        new_input['input_name'], str(e)))
+
+        model_inputs = input_model.get_inputs()
+        inputs_equal = True
+        if user_shapes:
+            # TODO: Remove this line when new 'cut' helper is introduced
+            if getattr(argv, "framework", None) != "pytorch":
+                # pytorch may use non-input nodes even when cutting is not used
+                raise_exception_for_input_output_cut(
+                    model_inputs, user_shapes, True)
+                inputs_equal = check_places_are_same(model_inputs, user_shapes)
             else:
-                value = np_map_cast[dtype](value)
-            value = np.array(value, dtype=dtype)
-
-            ov_shape = input_model.get_partial_shape(place)
-            if node.get('shape'):
-                # set user defined shape
-                ov_shape = PartialShape(node['shape'])
-                input_model.set_partial_shape(place, ov_shape)
-            elif ov_shape.is_dynamic:
-                # in case of dynamic shape (dynamic rank or dynamic dimension)
-                # deduce it based on the value shape and set it
-                ov_shape = PartialShape(value.shape)
-                input_model.set_partial_shape(place, ov_shape)
-
-            input_model.set_tensor_value(place, value)
-
-    def shape_to_array(shape: PartialShape):
-        return [shape.get_dimension(i) for i in range(shape.rank.get_length())]
+                inputs_equal = True
+
+        outputs_equal = True
+        if outputs:
+            # TODO: Remove this line when new 'cut' helper is introduced
+            if getattr(argv, "framework", None) != "pytorch":
```

Comment:
why do we check it if this is `else` branch corresponding to non-PyTorch case

Reply:
Agree, not needed